pinot-common/src/main/java/org/apache/pinot/pql/parsers/pql2/ast/ComparisonPredicateAstNode.java

```diff
@@ -155,6 +157,74 @@ public FilterQueryTree buildFilterQueryTree() {
     }
   }
 
+  @Override
+  public Expression buildFilterExpression() {
+    if (_identifier == null) {
+      throw new Pql2CompilationException("Comparison predicate has no identifier");
+    }
+
+    if ("=".equals(_operand)) {
+      if (_identifier != null && _literal != null) {
+        Expression expr = RequestUtils.createFunctionExpression(FilterKind.EQUALS.name());
+        expr.getFunctionCall().addToOperands(RequestUtils.createIdentifierExpression(_identifier));
+        expr.getFunctionCall().addToOperands(RequestUtils.createLiteralExpression(_literal));
+        return expr;
+      } else {
+        throw new Pql2CompilationException("Comparison is not between a column and a constant");
+      }
+    } else if ("<>".equals(_operand) || "!=".equals(_operand)) {
+      if (_identifier != null && _literal != null) {
+        Expression expr = RequestUtils.createFunctionExpression(FilterKind.NOT_EQUALS.name());
+        expr.getFunctionCall().addToOperands(RequestUtils.createIdentifierExpression(_identifier));
+        expr.getFunctionCall().addToOperands(RequestUtils.createLiteralExpression(_literal));
+        return expr;
+      } else {
+        throw new Pql2CompilationException("Comparison is not between a column and a constant");
+      }
+    } else {
+      if (_identifier != null) {
+        boolean identifierIsOnLeft = true;
```

Comment:
We add support to `1 < column`? Do we support this right now?
Also, how about function in comparison?